src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -692,7 +692,7 @@ def describe_trigger(parent: Candidate) -> str:
             + "the dependency conflict\n"
         )
 
-        logger.info(msg)
+        logger.critical(msg, extra={"color": "black"})
```

Comment:
Why should this be black? A lot of terminals have black/dark background. Having black on black wouldn't be consumable, no?

Reply:
Looking into this whole branchy structure, why not set a color variable dynamically and initialize `Style` once with that var?

Reply:
> Why should this be black?

It corresponds to ANSI 0: https://rich.readthedocs.io/en/stable/appendix/colors.html#appendix-colors

Reply:
> Looking into this whole branchy structure, why not set a color variable dynamically and initialize Style once with that var?

That can be done separately in a follow up. I'd like to not increase the scope of this PR, especially given that this is a first time contributor! :)

Reply:
> Why should this be black? A lot of terminals have black/dark background. Having black on black wouldn't be consumable, no?

Right, I thought the terminal emulator would deal with that if the background isn't white? Do we have a way to check the current terminal output color?

Reply:
Dunno, that's why I pointed it out. @pradyunsg do I understand correctly, that this is the color of the text, not the background?

Reply:
Yup. This will always be the default output color of the terminal.